### DISCUSSION TITLE
Additional Authentication Password Validator Settings

### DISCUSSION THREAD

**Discussion by mr1716** [netbox-community/netbox#15701] [Ideas]
Saw that AUTH_PASSWORD_VALIDATORS checks 4 things from DJANGO, but also curious if there would be a way to add in validations to ensure additional checks such as ensuring that passwords contain at least 1 upper case, 1 lower case, and 1 numeric value. I know that it checks for entirely numeric, and common values, plus 2 other checks. But adding in checks for character types would help improve security. The idea is that these would NOT be turned on and required by default, but configurable by the user.

I feel that this would really help improve the security of Netbox

**Comment by jeremystretch** [comment]
You can configure the [`AUTH_PASSWORD_VALIDATORS`](https://docs.netbox.dev/en/stable/configuration/security/#auth_password_validators) parameter however you'd like. See the [Django documentation](https://docs.djangoproject.com/en/5.0/topics/auth/passwords/#module-django.contrib.auth.password_validation) for details.

**Reply by mr1716** [reply]
@jeremystretch thanks. So it looks to be controlled by Django, so if I wanted to add new validators, then it would be a Django request, not Netbox?

**Reply by jeremystretch** [reply]
You can write your own custom validators; you just need to specify the dotted path to each when setting the parameter.

**Reply by mr1716** [reply]
Thanks. Much appreciated. I'll set the values in the configuration.py and keep the settings.py as "AUTH_PASSWORD_VALIDATORS = getattr(configuration, 'AUTH_PASSWORD_VALIDATORS', [])". But then how would I validate that the validator worked? Would I see the error message?
